packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,350 @@
+import normal from "@stdlib/stats/base/dists/normal";
+
+import {
+  PowerCalculationParams,
+  PowerCalculationResults,
+  MDEResults,
+  SampleSizeAndRuntime,
+  Week,
+} from "./types";
+
+/**
+ * delta method for relative difference
+ *
+ * @param varA Scalar control distribution variance.
+ * @param meanA Scalar control mean.
+ * @param nA Control sample size.
+ * @param varB Scalar treatment distribution variance.
+ * @param meanB Scalar treatment distribution mean.
+ * @param nB Treatment sample size.
+ * @param relative boolean indicator for relative effects.
+ * @returns variance.
+ */
+export function frequentistVariance(
+  varA: number,
+  meanA: number,
+  nA: number,
+  varB: number,
+  meanB: number,
+  nB: number,
+  relative: boolean,
+): number {
+  if (relative) {
+    return (
+      varB / (Math.pow(meanA, 2) * nB) +
+      (varA * Math.pow(meanB, 2)) / (Math.pow(meanA, 4) * nA)
+    );
+  } else {
+    return varB / nB + varA / nA;
+  }
+}
+
+export function powerStandardError(
+  variance: number,
+  mean: number,
+  n: number,
+  nVariations: number,
+  effectSize: number,
+): number {
+  return Math.sqrt(
+    frequentistVariance(
+      variance,
+      mean,
+      n / nVariations,
+      variance,
+      mean * (1 + effectSize),
+      n / nVariations,
+      true,
+    ),
+  );
+}
+
+export function calculateRho(
+  alpha: number,
+  sequentialTuningParameter: number,
+): number {
+  return Math.sqrt(
+    (-2 * Math.log(alpha) + Math.log(-2 * Math.log(alpha) + 1)) /
+      sequentialTuningParameter,
+  );
+}
+
+export function sequentialPowerSequentialVariance(
+  variance: number,
+  n: number,
+  alpha: number,
+  sequentialTuningParameter: number,
+): number {
+  const standardErrorSampleMean = Math.sqrt(variance / n);
+  const rho = calculateRho(alpha, sequentialTuningParameter);
+  const partUnderRadical =
+    (2 *
+      (n * Math.pow(rho, 2) + 1) *
+      Math.log(Math.sqrt(n * Math.pow(rho, 2) + 1) / alpha)) /
+    Math.pow(n * rho, 2);
+  const zSequential = Math.sqrt(n) * Math.sqrt(partUnderRadical);
+  const zStar = normal.quantile(1.0 - 0.5 * alpha, 0, 1);
+  const standardErrorSequential =
+    (standardErrorSampleMean * zSequential) / zStar;
+  const vSequential = Math.pow(standardErrorSequential, 2);
+  return n * vSequential;
+}
+
+export function sequentialPowerStandardError(
+  variance: number,
+  mean: number,
+  n: number,
+  nVariations: number,
+  effectSize: number,
+  alpha: number,
+  sequentialTuningParameter: number,
+): number {
+  const v = sequentialPowerSequentialVariance(
+    variance,
+    n / nVariations,
+    alpha,
+    sequentialTuningParameter,
+  );
+  return Math.sqrt(
+    frequentistVariance(
```

Comment:
I think the logic here is backwards. You need to compute the frequentistVariance and then use that  in `sequentialPowerSequentialVariance`. The crux is that the AsympCS method is applied to the treatment effect, not the variation means. 

That might elucidate why the `n` in your sequential adjustment actually needs to be `2 * n / nVariations`.

Reply:
agreed, changed